metaflow/cmd/code/__init__.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python
+import os
+import shutil
+import sys
+from subprocess import PIPE, run, CompletedProcess
+from tempfile import TemporaryDirectory
+from typing import List, Optional, Callable, Dict, Any
+
+from metaflow._vendor import click
+from metaflow import Run
+from metaflow.cli import echo_always
+
+EXCLUSIONS = [
+    "metaflow/",
+    "metaflow_extensions/",
+    "INFO",
+    "CONFIG_PARAMETERS",
+    "conda.manifest",
+]
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.group(help="Metaflow code commands")
+def code():
+    pass
+
+
+def echo(line: str) -> None:
+    echo_always(line, err=True, fg="magenta")
+
+
+def extract_code_package(runspec: str, exclusions: List[str]) -> TemporaryDirectory:
+    try:
+        run = Run(runspec, _namespace_check=False)
```

Comment:
nit: you are overriding the subprocess run which could lead to some confusion later.